crates/network-scanner/src/scanner.rs

```diff
@@ -393,7 +483,15 @@ pub struct NetworkScannerParams {
     pub netbios_timeout: u64,
     pub netbios_interval: u64,
     pub mdns_query_timeout: u64,
-    pub max_wait_time: u64, // max_wait for entire scan duration in milliseconds, suggested!
+    pub max_wait_time: u64,
+    pub ip_ranges: Vec<IpAddrRange>,
+}
+
+/// The parameters for configuring a network scanner. All fields are in milliseconds.
+#[derive(Debug, Clone, TypedBuilder)]
+pub struct NetworkScannerParams {
+    pub configs: ScannerConfig,
```

Comment:
naming: Typically not pluralized.
```suggestion
    pub config: ScannerConfig,
```